### PR TITLE
tests: Remove cross_staking mock

### PR DESCRIPTION
**Pull request by jawoznia**
part of #95

### CODE REVIEW

File: contracts/provider/external-staking/src/contract.rs [hunk 1]
```diff
@@ -492,12 +492,12 @@ impl ExternalStakingContract<'_> {
     /// Method used for tests only.
     #[msg(exec)]
     fn test_rollback_unstake(&self, ctx: ExecCtx, tx_id: u64) -> Result<Response, ContractError> {
```
[the remark this answers is not in the document]
Comment (jawoznia) [reply contracts/provider/external-staking/src/contract.rs:494]: I would love to get rid of them.
For now we could go with something like that: https://github.com/osmosis-labs/mesh-security/pull/117. Naming is bad I know.

The best workaround from `sylvia` side might be to add ability to conditionally use `messages` attribute.

Something like that would expose interface methods only if `mt` feature is enabled.
```rust
#[contract]
#[mt(messages(interface))]
```

[the remark this answers is not in the document]
Comment (JakeHartnell) [reply contracts/provider/external-staking/src/contract.rs:494]: Approach in #117 is definitely an improvement.

Love the idea of adding the ability to conditionally use `messages` attribute to `sylvia`.

File: contracts/provider/external-staking/Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "external-staking"
+name = "mesh-external-staking"
```
Comment (maurolacy) [inline contracts/provider/external-staking/Cargo.toml:2]: 👍🏼

File: contracts/provider/vault/src/multitest.rs [hunk 3]
```diff
@@ -451,12 +466,23 @@ fn stake_local() {
         .unwrap_err();
 }
 
+// Add comment
+// transaction ids are in different namespaces
+// if one transaction passes the second one cannot fail
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:471]:
```suggestion
// `vault` and `external-staking` transaction ids are in different ranges for clarity.
// The second (`vault`'s) transaction's commit or rollback cannot fail.
```

File: contracts/provider/vault/src/multitest.rs [hunk 4]
```diff
@@ -473,14 +499,10 @@ fn stake_cross() {
     // Contracts setup
 
     let local_staking_code = local_staking::multitest_utils::CodeId::store_code(&app);
-    let cross_staking_code = cross_staking::multitest_utils::CodeId::store_code(&app);
+    let cross_staking_code =
+        mesh_external_staking::contract::multitest_utils::CodeId::store_code(&app);
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:503]: 👍🏼

File: contracts/provider/vault/src/multitest.rs [hunk 5]
```diff
@@ -645,17 +736,75 @@ fn stake_cross() {
...
+    assert_eq!(
+        app.app()
+            .wrap()
+            .query_balance(&cross_staking.contract_addr, OSMO)
+            .unwrap(),
+        coin(0, OSMO)
+    );
+
+    // External staking contract will wait for `unbond_period` after receiving
+    // confirmation through the IBC channel that `unstake` was successfully executed.
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:778]: :+1 Good comment.

File: contracts/provider/vault/src/multitest.rs [hunk 6]
```diff
@@ -645,17 +736,75 @@ fn stake_cross() {
...
+    assert_eq!(
+        acc,
+        AccountResponse {
+            denom: OSMO.to_owned(),
+            bonded: Uint128::new(300),
+            free: ValueRange::new_val(Uint128::new(50)),
+        }
+    );
+
+    // After the unbounding period user can withdraw unbonded tokens
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:802]: Nit:
```suggestion
    // After the unbonding period user can withdraw unbonded tokens
```

File: contracts/provider/vault/src/multitest.rs [hunk 7]
```diff
@@ -703,15 +855,15 @@ fn stake_cross() {
         AccountResponse {
             denom: OSMO.to_owned(),
             bonded: Uint128::new(300),
-            free: ValueRange::new_val(Uint128::new(200)),
+            free: ValueRange::new_val(Uint128::new(100)),
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:858]: Second unstake not unbonded yet. Good tests.

File: contracts/provider/vault/src/multitest.rs [hunk 8]
```diff
@@ -734,11 +886,7 @@ fn stake_cross() {
     // Error not verified as it is swallowed by intermediate contract
     // int this scenario
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:887]:
```suggestion
    // in this scenario
```

File: contracts/provider/vault/src/multitest.rs [hunk 9]
```diff
@@ -919,7 +1096,7 @@ fn stake_cross_txs() {
             .unwrap()
             .claims,
         [LienResponse {
-            lienholder: "contract0".to_string(),
+            lienholder: "contract2".to_string(),
```
Comment (maurolacy) [inline contracts/provider/vault/src/multitest.rs:1099]: Different instantiation order, right?

**Review by maurolacy — APPROVED** [review]
LGTM. Great work. Thanks for going through this.

File: contracts/provider/vault/src/multitest.rs (hunk 9 above)
Comment (jawoznia) [reply contracts/provider/vault/src/multitest.rs:1099]: exactly

**Pull request merged**